list.go

```diff
@@ -0,0 +1,31 @@
+package ctxvfs
+
+import "context"
+
+// ListAllFileser is type that implements ListAllFiles
+type ListAllFileser interface {
+	// ListAllFiles returns a slice of all file paths on a VFS. This is
+	// only files (excludes directories). This interface exists for
+	// implementors to provide a faster way to list all files than walking
+	// the file tree with ReadDir/etc.
+	ListAllFiles(ctx context.Context) ([]string, error)
+}
+
+// ListAllFiles returns a slice of all file paths (excludes directories).
+func ListAllFiles(ctx context.Context, fs FileSystem) ([]string, error) {
```

Comment:
I agree, but there are a few VFSes we use only in test environments, and I'd rather not make them all implement the fast path. What do you suggest, since the fallback is legit for that use case.